tests/func/metrics/test_show.py

```diff
@@ -110,3 +110,31 @@ def test_missing_cache(tmp_dir, dvc, run_copy_metrics):
     remove(stage.outs[0].cache_path)
 
     assert dvc.metrics.show() == {"": {"metrics.yaml": 1.1}}
+
+
+def test_show_non_metric(tmp_dir, dvc):
+    tmp_dir.gen("metrics.yaml", "foo: 1.1")
+
+    assert dvc.metrics.show(targets=["metrics.yaml"]) == {
+        "": {"metrics.yaml": {"foo": 1.1}}
+    }
+
+
+def test_show_non_metric_and_recursive_metric(tmp_dir, dvc, run_copy_metrics):
+    tmp_dir.gen({"metrics_t.yaml": "foo: 1.1", "metrics": {}})
+
+    metric1 = os.fspath(tmp_dir / "metrics" / "metric1.yaml")
+    metric2 = os.fspath(tmp_dir / "metrics" / "metric2.yaml")
+    run_copy_metrics("metrics_t.yaml", metric1, metrics=[metric1])
+    run_copy_metrics("metrics_t.yaml", metric2, metrics=[metric2])
```

Comment:
These are not really related to this PR, right? Or did you want to test if recursive works for dirs that don't have official metrics? Just a bit confused.

Reply:
I wanted to verify whether `show` will work as intended if we mix `non-metric` and `metric` approach in case of `recursive=true`

Reply:
Makes sense.